R/wrappers.R

```diff
@@ -6,6 +6,7 @@ fcoalesce   = function(...) .Call(Ccoalesce, list(...), FALSE)
 setcoalesce = function(...) .Call(Ccoalesce, list(...), TRUE)
 
 fifelse = function(test, yes, no, na=NA) .Call(CfifelseR, test, yes, no, na)
+fcase   = function(..., default=NA) .Call(CfcaseR, default, parent.frame(), as.list(substitute(...())))
```

Comment:
what is the purpose of `...()` rather than `...`?

Reply:
The goal here is that i want to pass a list to the C function, if I don't use as.list what is passed is a pair list and if I don't use parentheses it will just be a ling expression ig i remember well. You can create a simple function with the different option and see the output. I am adding links that might be helpful:
https://stat.ethz.ch/R-manual/R-devel/library/base/html/Paren.html

Reply:
First reference you provide Paren.html does not seems to be related to this use case. It would be related if you would use `(...)` and not `...()`.

Reply:
```r
is.call(quote(...))
#[1] FALSE
is.call(quote(...()))
#[1] TRUE
```
```r
as.list(quote(...()))
#[[1]]
#...

as.list(quote(...))
#[[1]]
#...
```

Reply:
did you tried `substitute(list(...))` instead?

Reply:
I have tried many combinations. I probably tried it yes. I might need to retry it and come back to you. But yes I agree if it behave the same we should use it because it is easier to read.

Reply:
R uses `.()` inside `bquote`, and recently (still in R-devel) it also has `..()` use cases there. Thus using `...()` might be confusing, or eventually even break if R parses catches that somehow different, if ever R's introduce `...()` use too.

Reply:
`list(substitute(...))` does not work it is not the same  as the one current used. It seems to be treated differently at C level. Give it a try if you can. If you have another alternative I open to suggestions.

Reply:
@2005m 
https://stat.ethz.ch/pipermail/r-devel/2020-May/079413.html
interesting insight into this subject by an R core member